xdsl/dialects/mpi.py

```diff
@@ -775,6 +775,63 @@ def get(req: SSAValue | Operation):
         return NullRequestOp.build(operands=[req])
 
 
+@irdl_op_definition
+class GatherOp(MPIBaseOp):
+    """
+    This is used to gather data into one big buffer.
+
+    int MPI_Gather(const void *sendbuf, int sendcount, MPI_Datatype sendtype,
+                    void *recvbuf, int recvcount, MPI_Datatype recvtype,
+                    int root,
+                    MPI_Comm comm)
+
+     - sendbuf, sendcount, sendtype: info on the buffer to be sent
+     - recvbuf, recvcount, recvtype: info on the gather buffer
+     - root: the rank that receives all the data
+     - comm: the communicator to use
+
+    Note: recvcount * sizeof(recvtype) == sendcount * sizeof(sendtype) * N
+    where N is the number of nodes participating in the gather.
```

Comment:
Can you somehow enforce this in a verifier? I am guessing that you are missing N, right?

Reply:
We are missing N, and it's all SSA values right?

Reply:
Ah, true. Why is there no facepalm on git? 🤔